### PR TITLE
Add sol parser

### PR DESCRIPTION
**Pull request by KirillDogadin-std**

### CODE REVIEW

**Comment by KirillDogadin-std** [comment]
On etherscan "when looking at the src code" there're always 4 lines on top that do not end up in the database when we download it:

```
1. /**
2.  *Submitted for verification at Etherscan.io on 2023-08-03
3. */
4.
```

**Comment by KirillDogadin-std** [comment]
Tested on:

http://localhost:3000/api/linking/goerli/0x4Ca8451B5a598aba0160F35505402fad9713fb0c

seems to work nice

File: app/src/helpers/getContractAddresses.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { Contract } from '@prisma/client';
+import { parse, visit } from '@solidity-parser/parser'
+import {AddressInfo} from '~/types'
+
+const isAddress = (val: string) => {
```
Comment (DaeunYoon) [inline app/src/helpers/getContractAddresses.ts:5]: You can use the [`viem` function](https://viem.sh/docs/utilities/isAddress.html) here as well

**Comment by DaeunYoon** [comment]
http://localhost:3000/api/linking/goerli/0x87d40a7D994c98F44720D2d960752E37db4569D0
http://localhost:3000/api/linking/goerli/0x7dd0e3d2d740c6fc08ed2e2bd4122f239ad77875

parsing seems to work! :D

**Comment by valiafetisov** [comment]
Also works for me! But can you please add actual addresses to the output under `address`? Or do you plan to do it in the next PR/iteration?

**Comment by KirillDogadin-std** [comment]
Done

**Review by valiafetisov — APPROVED** [review]

**Pull request merged**